js/src/personal/haoking/Haoking.page.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  MantineProvider,
+  createTheme,
+  Title,
+  Text,
+  Button,
+  Box,
+  AppShell,
+  Flex,
+  Image,
+} from '@mantine/core'
+import { IconBrandGithub, IconBrandLinkedin } from '@tabler/icons-react'
+import styles from './Haoking.module.css'
+import { useState } from 'react'
+
+const darkTheme = createTheme({
+  colors: {
+    black: [
+      '#6b6b6b',
+      '#545454',
+      '#363636',
+      '#262626',
+      '#212121',
+      '#202020',
+      '#101010',
+      '#303030',
+      '#1f1f1f',
+      '#1a1a1a',
+    ],
+    blue: [
+      '#7472db',
+      '#4536f7',
+      '#483fa1',
+      '#202378',
+      '#0e2159',
+      '#6484e3',
+      '#786dfc',
+      '#0b2f9c',
+      '#051440',
+      '#1b2540',
+    ],
+  },
+  fontFamily: 'Helvetica, sans-serif',
+})
+
+export function HaokingPage() {
+  // 0 is light, 1 is dark
+  const [colorState, setColor] = useState(0)
+
+  return (
+    <>
+      <MantineProvider theme={darkTheme}>
```

Comment:
I'm a little bit concerned with having nested MantineProviders. Don't know exactly how it works, but this one may effect the root one. 

May be good to read through this thread. 
https://github.com/orgs/mantinedev/discussions/4803

You might need to set getRootElement here too?